ctpolicy/loglist/loglist.go

```diff
@@ -30,70 +41,80 @@ type LogList map[string]operatorGroup
 
 // operatorGroup is a convenience type allowing constant-time lookup of logs
 // within a group by their unique ID.
-type operatorGroup map[string]*generated.LogListSchemaJsonOperatorsElemLogsElem
-
-// New returns a LogList built from the autogenerated and embedded JSON blob.
-func New() (LogList, error) {
-	if !logList.ready {
-		logList.Lock()
-		defer logList.Unlock()
+type operatorGroup map[string]logInfo
+
+// logInfo is a convenience type allowing certain fields on the auto-generated
+// generated.LogListSchemaJsonOperatorsElemLogsElem to be pre-parsed and stored.
+type logInfo struct {
+	Url            string
+	Key            string
+	startInclusive time.Time
+	endExclusive   time.Time
+}
 
-		err := json.Unmarshal(generated.LogListJSON, &logList)
-		if err != nil {
-			return nil, err
+// contains is a simple helper to see if an element is in a slice, since we
+// don't have generic stdlib functions for this yet.
+func contains(haystack []ctconfig.LogID, needle string) bool {
+	for _, item := range haystack {
+		if item.ID == needle {
+			return true
 		}
+	}
+	return false
+}
 
-		logList.ready = true
+// New returns a LogList which is a subset of the autogenerated and embedded
+// JSON blob, containing only those logs whose LogIDs match the given list of
+// IDs. If no log IDs are provided, all logs are presumed to match.
```

Comment:
The "no log IDs" special case is the reverse of what I'd expect. I'd expect none to match in that case.

It looks like that special case is used in just one place: the lint. What about making a separate `AllLogs()` function that returns them all? They could both delegate to a private function to allow shared implementation.